From ce463daec9da0f52a53fe738aa70e2ae4e4ecb13 Mon Sep 17 00:00:00 2001
From: Jacky Chou <jacky_chou@aspeedtech.com>
Date: Tue, 5 Aug 2025 16:11:16 +0800
Subject: [PATCH 04/32] pinctrl: soc1: add a group for RMII RCLK OUT

This pin is not in RMII pin.
50MHz clock is provided by PHY.
Therefore, split it from RMII pinctrl.
Here adds a new group for RMII RCLK OUT.
If you need to use internal 50MHz clock, add this pintctrl
into to your dts.

Signed-off-by: Jacky Chou <jacky_chou@aspeedtech.com>
Change-Id: I9ff24a8428f29b90bbbff40c3a62c7befe85f66d
(cherry picked from commit a859f15f38a15aeb3a1a074062e1c6e2bf158653)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c | 22 +++++++++++++++----
 1 file changed, 18 insertions(+), 4 deletions(-)

diff --git a/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c b/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c
index a57f0a0963..68640264e2 100644
--- a/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c
+++ b/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c
@@ -89,8 +89,10 @@ static struct ast2700_soc1_sig_desc rgmii1_link[] = {
 };
 
 static struct ast2700_soc1_sig_desc rmii0_link[] = {
-	{ 0x44, GENMASK(31, 0),
-	  BIT(1) | BIT(9) | BIT(13) | BIT(17) | BIT(21) | BIT(25) | BIT(29) },
+	{ 0x44,
+	  GENMASK(30, 28) | GENMASK(22, 20) | GENMASK(18, 16) |
+	  GENMASK(14, 12) | GENMASK(10, 8) | GENMASK(2, 0),
+	  BIT(1) | BIT(9) | BIT(13) | BIT(17) | BIT(21) | BIT(29) },
 	{ 0x48, GENMASK(6, 0), BIT(1) | BIT(5) },
 	/* IO Driving */
 	{ 0xD4, GENMASK(23, 0),
@@ -98,9 +100,15 @@ static struct ast2700_soc1_sig_desc rmii0_link[] = {
 		  BIT(18) }
 };
 
+static struct ast2700_soc1_sig_desc rmii0rclko_link[] = {
+	{ 0x44, GENMASK(26, 24), BIT(25) },
+};
+
 static struct ast2700_soc1_sig_desc rmii1_link[] = {
-	{ 0x4c, GENMASK(31, 0),
-	  BIT(1) | BIT(9) | BIT(13) | BIT(17) | BIT(21) | BIT(25) | BIT(29) },
+	{ 0x4c,
+	  GENMASK(30, 28) | GENMASK(22, 20) | GENMASK(18, 16) |
+	  GENMASK(14, 12) | GENMASK(10, 8) | GENMASK(2, 0),
+	  BIT(1) | BIT(9) | BIT(13) | BIT(17) | BIT(21) | BIT(29) },
 	{ 0x50, GENMASK(6, 0), BIT(1) | BIT(5) },
 	/* IO Driving */
 	{ 0xD8, GENMASK(23, 0),
@@ -108,6 +116,10 @@ static struct ast2700_soc1_sig_desc rmii1_link[] = {
 		  BIT(18) }
 };
 
+static struct ast2700_soc1_sig_desc rmii1rclko_link[] = {
+	{ 0x4c, GENMASK(26, 24), BIT(25) },
+};
+
 static struct ast2700_soc1_sig_desc sgmii_link[] = {
 	{0x7c, BIT(0), BIT(0)},
 };
@@ -340,7 +352,9 @@ static const struct aspeed_group_config ast2700_soc1_groups[] = {
 	{ "RGMII0", ARRAY_SIZE(rgmii0_link), rgmii0_link },
 	{ "RGMII1", ARRAY_SIZE(rgmii1_link), rgmii1_link },
 	{ "RMII0", ARRAY_SIZE(rmii0_link), rmii0_link },
+	{ "RMII0RCLKO", ARRAY_SIZE(rmii0rclko_link), rmii0rclko_link },
 	{ "RMII1", ARRAY_SIZE(rmii1_link), rmii1_link },
+	{ "RMII1RCLKO", ARRAY_SIZE(rmii1rclko_link), rmii1rclko_link },
 	{ "SGMII", ARRAY_SIZE(sgmii_link), sgmii_link },
 	{ "FWSPIQUAD", ARRAY_SIZE(fwspiquad_link), fwspiquad_link },
 	{ "SPI0", ARRAY_SIZE(spi0_link), spi0_link },
-- 
2.34.1

